lib/jsonapi-serializers/dynamic_proxy_object.rb

```diff
@@ -0,0 +1,24 @@
+module JSONAPI
+	class DynamicProxyObject < BasicObject
+
+    def initialize(target)
+      @cache = {}
+      @target = target
+    end
+
+    def class
+      @target.class
+    end
+
+    def cache_key(name, args)
+      "#{name}_#{args.hash}"
+    end
+
+    def method_missing(name, *args, &block)
+      key = cache_key(name, args)
+      @cache[key] ||= @target.send(name, *args, &block)
```

Comment:
This should be `public_send`, for maximum clarity.

Reply:
I don't think `||=` will be a good enough check. Different args or a block should produce different results.